component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -484,7 +484,25 @@ private void processSMSOTPFlow(AuthenticationContext context, HttpServletRequest
                 String loginPage = SMSOTPUtils.getMobileNumberRequestPage(context);
                 try {
                     String url = getURL(loginPage, queryParams);
-                    response.sendRedirect(url);
+                    String mobileNumberPatternViolationIssue = SMSOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_VIOLATED;
+                    if (StringUtils.isNotEmpty(
+                            context.getAuthenticatorProperties().get(SMSOTPConstants.MOBILE_NUMBER_REGEX))) {
+                        // Check for regex is violation error message configured in idp configuration.
+                        if (StringUtils.isNotEmpty(context.getAuthenticatorProperties()
+                                .get(SMSOTPConstants.MOBILE_NUMBER_PATTERN_FAILURE_ERROR_MESSAGE))) {
+                            mobileNumberPatternViolationIssue = context.getAuthenticatorProperties()
+                                    .get(SMSOTPConstants.MOBILE_NUMBER_PATTERN_FAILURE_ERROR_MESSAGE);
+                        }
+                        // Send the response with encoded regex pattern and error message.
+                        response.sendRedirect(url + SMSOTPConstants.MOBILE_NUMBER_REGEX_PATTERN_QUERY +
```

Comment:
url might not have query params. Better to use FrameworkUtils.appendQueryParam method.